### PR TITLE
Change of name dob detail page

### PR DESCRIPTION
**Pull request by hortha**
### Context

Support agents need to be able to see the details a Change of name/DOB request so that they can approve or decline it.

### Changes proposed in this pull request

Add a new page at /cases/{id} following the figma designs.

If the specified case does not exist, return a 404.

If the specified case’s state is not Active, return a 400.

The page header should be the case subject’s title.

The case reference, person name and ‘created on’ date should match what’s in the case list table.

For a change of name request, show the current and requested first, middle and last names in a summary list. For a change of DOB request, show the current and requested date of birth.

The Proof section should show every document attached to the case (there could be multiple) with a ‘View larger’ link. The view larger URL should be /cases/{id}/documents/{document ID}.

The Accept change and Reject change buttons should link to /cases/{id}/accept and /cases/{id}/reject, respectively.

The Back link should go to the case list page.

### Guidance to review

Changes to CRM queries, support UI pages + tests

### Checklist

-   [x] Attach to Trello card
-   [x] Rebased master
-   [ ] Cleaned commit history
-   [x] Tested by running locally

### CODE REVIEW

**Comment by hortha** [comment]
[Change of name/DOB detail page](https://trello.com/c/QCSwApCG/1179-change-of-name-dob-detail-page)

File: TeachingRecordSystem/src/TeachingRecordSystem.Core/Dqt/QueryHandlers/GetIncidentByTicketNumberHandler.cs [hunk 1]
```diff
@@ -0,0 +1,121 @@
+using Microsoft.PowerPlatform.Dataverse.Client;
+using Microsoft.Xrm.Sdk.Query;
+using TeachingRecordSystem.Core.Dqt.Queries;
+
+namespace TeachingRecordSystem.Core.Dqt.QueryHandlers;
+public class GetIncidentByTicketNumberHandler : ICrmQueryHandler<GetIncidentByTicketNumberQuery, (Incident?, dfeta_document[]?)>
```
Comment (gunndabad) [inline TeachingRecordSystem/src/TeachingRecordSystem.Core/Dqt/QueryHandlers/GetIncidentByTicketNumberHandler.cs:6]: Nit:
```suggestion
namespace TeachingRecordSystem.Core.Dqt.QueryHandlers;

public class GetIncidentByTicketNumberHandler : ICrmQueryHandler<GetIncidentByTicketNumberQuery, (Incident?, dfeta_document[]?)>
```

File: TeachingRecordSystem/src/TeachingRecordSystem.Core/Dqt/Queries/GetIncidentByTicketNumberQuery.cs [hunk 2]
```diff
@@ -0,0 +1,3 @@
+namespace TeachingRecordSystem.Core.Dqt.Queries;
+
+public record GetIncidentByTicketNumberQuery(string TicketNumber) : ICrmQuery<(Incident?, dfeta_document[]?)>;
```
[the remark this answers is not in the document]
Comment (hortha) [reply TeachingRecordSystem/src/TeachingRecordSystem.Core/Dqt/Queries/GetIncidentByTicketNumberQuery.cs:3]: Is that right? The tuple is never null but the individual properties can be

File: TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Pages/Cases/EditCase/Index.cshtml [hunk 3]
```diff
@@ -1,5 +1,153 @@
...
+                    }
+                </ul>
+            </div>
+        </div>
+    </div>
+</div>
+
+<div class="govuk-button-group">
+    <a href="@LinkGenerator.AcceptCase(Model.TicketNumber)" class="govuk-button">Accept change</a>
+    <a href="@LinkGenerator.RejectCase(Model.TicketNumber)" class="govuk-button govuk-button--warning">Reject change</a>
```
Comment (gunndabad) [inline TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Pages/Cases/EditCase/Index.cshtml:152]:
```suggestion
    <govuk-button-link href="@LinkGenerator.AcceptCase(Model.TicketNumber)">Accept change</govuk-button-link>
    <govuk-button-link href="@LinkGenerator.RejectCase(Model.TicketNumber)" class="govuk-button--warning">Reject change</govuk-button-link>
```

File: TeachingRecordSystem/src/TeachingRecordSystem.Core/Dqt/Queries/GetIncidentByTicketNumberQuery.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (gunndabad) [reply TeachingRecordSystem/src/TeachingRecordSystem.Core/Dqt/Queries/GetIncidentByTicketNumberQuery.cs:3]: Presumably you'd never get a `null` Incident and non-`null` `dfeta_document[]` (or vice-versa)?

File: TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Pages/Shared/_Layout.cshtml [hunk 4]
```diff
@@ -1,7 +1,18 @@
...
+@inject IAuthorizationService AuthorizationService
 @{
     Layout = "_GovUkPageTemplate";
 }
 
+@section Head {
+    @PageTemplateHelper.GenerateStyleImports()
+    <link rel="stylesheet" asp-href-include="~/Styles/site.css">
+        @RenderSection("Styles", required: false)
+        @RenderSection("Scripts", required: false)
```
Comment (gunndabad) [inline TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Pages/Shared/_Layout.cshtml:13]:
```suggestion
    @RenderSection("Styles", required: false)
    @RenderSection("Scripts", required: false)
```

File: TeachingRecordSystem/tests/TeachingRecordSystem.Core.Dqt.Tests/QueryTests/GetDocumentByIdTests.cs [hunk 5]
```diff
@@ -0,0 +1,44 @@
+namespace TeachingRecordSystem.Core.Dqt.Tests.QueryTests;
+public class GetDocumentByIdTests : IAsyncLifetime
```
Comment (gunndabad) [inline TeachingRecordSystem/tests/TeachingRecordSystem.Core.Dqt.Tests/QueryTests/GetDocumentByIdTests.cs:2]:
```suggestion
namespace TeachingRecordSystem.Core.Dqt.Tests.QueryTests;

public class GetDocumentByIdTests : IAsyncLifetime
```

File: TeachingRecordSystem/tests/TeachingRecordSystem.SupportUi.Tests/PageTests/Cases/EditCase/IndexTests.cs [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+
+namespace TeachingRecordSystem.SupportUi.Tests.PageTests.Cases.EditCase;
+
+[Collection(nameof(DisableParallelization))]
+public class IndexTests : TestBase
+{
+    public IndexTests(HostFixture hostFixture)
+        : base(hostFixture)
+    {
+        XrmFakedContext.DeleteAllEntities<Incident>();
```
Comment (gunndabad) [inline TeachingRecordSystem/tests/TeachingRecordSystem.SupportUi.Tests/PageTests/Cases/EditCase/IndexTests.cs:11]: Do we need this? (If not `[Collection(nameof(DisableParallelization))]` can go too.)

**Review by gunndabad — APPROVED** [review]

File: TeachingRecordSystem/src/TeachingRecordSystem.Core/Dqt/Queries/GetIncidentByTicketNumberQuery.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (hortha) [reply TeachingRecordSystem/src/TeachingRecordSystem.Core/Dqt/Queries/GetIncidentByTicketNumberQuery.cs:3]: I'm sure DQT database wise it is technically possible but not when these cases are created via the API .. i.e. either they will both be null or both not null .. does that mean what I already have is acceptable ?

[the remark this answers is not in the document]
Comment (hortha) [reply TeachingRecordSystem/src/TeachingRecordSystem.Core/Dqt/Queries/GetIncidentByTicketNumberQuery.cs]: Done

File: TeachingRecordSystem/src/TeachingRecordSystem.Core/Dqt/QueryHandlers/GetIncidentByTicketNumberHandler.cs (hunk 1 above)
Comment (hortha) [reply TeachingRecordSystem/src/TeachingRecordSystem.Core/Dqt/QueryHandlers/GetIncidentByTicketNumberHandler.cs]: Done

File: TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Pages/Shared/_Layout.cshtml (hunk 4 above)
Comment (hortha) [reply TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Pages/Shared/_Layout.cshtml]: Done

File: TeachingRecordSystem/tests/TeachingRecordSystem.Core.Dqt.Tests/QueryTests/GetDocumentByIdTests.cs (hunk 5 above)
Comment (hortha) [reply TeachingRecordSystem/tests/TeachingRecordSystem.Core.Dqt.Tests/QueryTests/GetDocumentByIdTests.cs]: Done

File: TeachingRecordSystem/tests/TeachingRecordSystem.SupportUi.Tests/PageTests/Cases/EditCase/IndexTests.cs (hunk 6 above)
Comment (hortha) [reply TeachingRecordSystem/tests/TeachingRecordSystem.SupportUi.Tests/PageTests/Cases/EditCase/IndexTests.cs]: Removed

**Pull request merged**